homepage/homepage/content/docs/groups/inheritance.mdx

```diff
@@ -1,153 +1,190 @@
 
-// Bob loses access to both parent and child groups
+// Works only if I'm a member of `companyGroup`
+teamGroup.addMember(companyGroup);
 ```
 </CodeGroup>
 
-## Role Inheritance Rules
+### Overriding parent group roles
+
+In some cases you might want to inherit all members from a parent group but override their roles to the same specific role in the child group. You can do so by passing an "override role" as a second argument to `addMember`:
 
-If the account is already a member of the child group, it will get the more permissive role:
 <CodeGroup>
-```typescript
-const parentGroup = Group.create();
-parentGroup.addMember(bob, "reader");
+```ts twoslash
+import { Group } from "jazz-tools";
+import { createJazzTestAccount } from 'jazz-tools/testing';
+const bob = await createJazzTestAccount();
+// ---cut---
+const organizationGroup = Group.create();
+organizationGroup.addMember(bob, "admin");
 
-const childGroup = Group.create();
-parentGroup.addMember(bob, "writer");
-childGroup.extend(parentGroup);
+const billingGroup = Group.create();
 
-// Bob stays a writer because his role is higher
-// than the inherited reader role.
+// This way the members of the organization
+// can only read the billing data
+billingGroup.addMember(organizationGroup, "reader");
 ```
 </CodeGroup>
 
-When extending groups, only admin, writer and reader roles are inherited:
+The "override role" works in both directions:
+
 <CodeGroup>
-```typescript
+```ts twoslash
+import { Group } from "jazz-tools";
+import { createJazzTestAccount } from 'jazz-tools/testing';
+const alice = await createJazzTestAccount();
+const bob = await createJazzTestAccount();
+// ---cut---
 const parentGroup = Group.create();
-parentGroup.addMember(bob, "writeOnly");
+parentGroup.addMember(bob, "reader");
+parentGroup.addMember(alice, "admin");
 
 const childGroup = Group.create();
-childGroup.extend(parentGroup);
+childGroup.addMember(parentGroup, "writer");
 
-// Bob does not become a member of the child group
+// Bob and Alice are now writers in the child group
 ```
 </CodeGroup>
 
-To extend a group:
+### Permission changes
 
-1. The current account must be an admin in the child group
-2. The current account must be a member of the parent group
+When you remove a member from a parent group, they automatically lose access to all child groups. We handle key rotation automatically to ensure security.
 
 <CodeGroup>
-```typescript
-const companyGroup = company._owner.castAs(Group)
-const teamGroup = Group.create();
+```ts twoslash
+import { Group } from "jazz-tools";
+import { createJazzTestAccount } from 'jazz-tools/testing';
+const bob = await createJazzTestAccount();
+const parentGroup = Group.create();
+parentGroup.addMember(bob, "reader");
+// ---cut---
+// Remove member from parent
+await parentGroup.removeMember(bob);
 
-// Works only if I'm a member of companyGroup
-teamGroup.extend(companyGroup); 
+// Bob loses access to both parent and child groups
 ```
```

Comment:
This might be a good opportunity to also explain that removing a member from the container group doesn't revoke access if they also have access to a member group.